src/lib/__tests__/routing-test.ts

```diff
@@ -0,0 +1,152 @@
+import { createSearchClient } from '../../../test/mock/createSearchClient';
+import { wait } from '../../../test/utils/wait';
+import historyRouter from '../routers/history';
+import instantsearch from '../..';
+import { connectSearchBox } from '../../connectors';
+
+describe('routing', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('writeOnDispose=true', () => {
+    test('cleans URL on dispose', async () => {
+      const pushState = jest.spyOn(window.history, 'pushState');
+      const writeDelay = 400;
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient: createSearchClient(),
+        routing: {
+          router: historyRouter({
+            writeDelay,
+          }),
+        },
+      });
+
+      search.addWidgets([connectSearchBox(() => {})({})]);
+
+      search.start();
+
+      // Wait for ${writeDelay} to pass then check URL has been initialized
+      await wait(1.5 * writeDelay);
+      expect(window.location.search).toEqual('');
+      expect(pushState).toHaveBeenCalledTimes(0);
+
+      // Trigger an update - push a change
+      search.renderState.indexName!.searchBox!.refine('Apple');
+
+      // Wait for ${writeDelay} to pass then check URL has been updated
+      await wait(1.5 * writeDelay);
+      expect(window.location.search).toEqual(
+        `?${encodeURI('indexName[query]=Apple')}`
+      );
+      expect(pushState).toHaveBeenCalledTimes(1);
+
+      // Trigger a dispose
+      search.dispose();
+
+      // Wait for ${writeDelay} to pass then check URL has been cleaned
+      await wait(1.5 * writeDelay);
```

Comment:
I'm assuming the 1.5 factor is to ensure the `writeDelay` did elapse and avoid false positives. Since we run this 10 times from what I've counted, we're adding 6 seconds to this test suite.

Could we reduce `writeDelay` to avoid slowing down the test suite too much?